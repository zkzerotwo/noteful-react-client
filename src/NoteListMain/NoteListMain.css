.NoteListMain {
    background-color: #efefef;
    height: 100%;
    padding: 10px;
    width: 100%;
}

.NoteListMain__content {
    color: #3a3939;
    max-width: 380px;
}