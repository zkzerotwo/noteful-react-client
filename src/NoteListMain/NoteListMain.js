import React from 'react'
import Notes from '../Notes/Notes'
import NotefulContext from '../NotefulContext'
import PropTypes from 'prop-types'
import { findNote } from '../folderFinders'
import './NoteListMain.css'

class NoteListMain extends React.Component {
    static defaultProps = {
        history: {
            goBack: () => { }
        },
        match: {
            params: {}
        }
    }
    handleDeleteNote = noteId => {
        this.props.history.push(`/`)
    }

    static contextType = NotefulContext

    render() {
        const { notes = [] } = this.context
        const { noteId } = this.props.match.params
        console.log(notes, noteId, "looking for notes")
        const note = findNote(notes, noteId) || { content: '' }
        console.log(notes, "notes in context")
        return (
            <section className='NoteListMain'>
                <Notes
                    id={note.id}
                    name={note.name}
                    modified={note.modified}
                    onDeleteNote={this.handleDeleteNote}
                />
                <div className='NoteListMain__content'>
                    {note.content.split(/\n \r|\n/).map((para, i) =>
                        <p key={i}>{para}</p>
                    )}
                </div>
            </section>
        )
    }
}
export default NoteListMain;

NoteListMain.propTypes = {
    notes: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
        content: PropTypes.string.isRequired,
        folderId: PropTypes.string.isRequired,
        modified: PropTypes.string.isRequired
    }))
}