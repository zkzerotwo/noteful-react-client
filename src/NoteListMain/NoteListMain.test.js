import React from 'react';
import { shallow } from 'enzyme'
import  Enzyme  from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import NoteListMain from './NoteListMain'

Enzyme.configure({ adapter: new Adapter() });
describe(`NoteListMain component`, () => {
    const props = {
        note: {
            "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
            "name": "Dogs",
            "modified": "2019-01-03T00:00:00.000Z",
            // "folderId": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
            "content": "Corporis accusamus placeat.\n \rUnde."
        }
    }

    it('renders a .NoteListMain by default', () => {
        const wrapper = shallow(<NoteListMain />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders a Note with note prop', () => {
        const note = shallow(<NoteListMain {...props} />)
            .find('Note')
        expect(toJson(note)).toMatchSnapshot()
    })

    it(`splits the content by \\n or \\n\\r, with a p foreach`, () => {
        [{
            note: { "content": "Content with n r.\n \rafter n r." }
        }, {
            note: { "content": "Content with n.\nafter." }
        }].forEach(props => {
            const content = shallow(<NoteListMain {...props} />)
                .find('NoteListMain__content')
            expect(toJson(content)).toMatchSnapshot()
        })
    })
})