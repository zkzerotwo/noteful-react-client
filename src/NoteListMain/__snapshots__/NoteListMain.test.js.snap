// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteListMain component renders a .NoteListMain by default 1`] = `
<section
  className="NoteListMain"
>
  <Notes
    onDeleteNote={[Function]}
  />
  <div
    className="NoteListMain__content"
  >
    <p
      key="0"
    />
  </div>
</section>
`;

exports[`NoteListMain component renders a Note with note prop 1`] = `null`;

exports[`NoteListMain component splits the content by \\n or \\n\\r, with a p foreach 1`] = `null`;

exports[`NoteListMain component splits the content by \\n or \\n\\r, with a p foreach 2`] = `null`;
